src/assignment3/Assignment3.java

```diff
@@ -0,0 +1,67 @@
+package assignment3;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Assignment3 {
+
+    private static void recursiveSearch(char[][] grid, boolean visited[][],
+            int width, int height, int startI, int startJ,
+            Dictionary dict, String prefix, Set<String> foundWords) {
+        
+        visited[startI][startJ] = true;
+        if (dict.isWord(prefix)) {
+            if (!foundWords.contains(prefix)) {

```

Comment:
No need to check if result is already included. Can you please remove it? It's usually better to avoid unnecessary code unless it has clear readability benefits, which is not the case here.